testing/oauth_server.ts

```diff
@@ -33,30 +40,57 @@ export function launchOAuthServerFlow({
   const {authorizationUrl, tokenUrl, additionalParams, scopeDelimiter} = authDef;
   // Use the manifest's scopes as a default.
   const requestedScopes = scopes && scopes.length > 0 ? scopes : authDef.scopes;
-  // ClientOAuth2 doesn't support a scope delimiter, so just hack a fake single pre-delimited scope
-  // if there's a custom delimiter involved.
-  const scopeArray =
-    requestedScopes && scopeDelimiter && scopeDelimiter !== ' '
-      ? [requestedScopes.join(scopeDelimiter)]
-      : requestedScopes;
-  const oauth2Client = new ClientOAuth2({
-    clientId,
-    clientSecret,
-    authorizationUri: authorizationUrl,
-    accessTokenUri: tokenUrl,
-    scopes: scopeArray,
-    redirectUri: makeRedirectUrl(port),
-    query: additionalParams,
+  const scope = requestedScopes ? requestedScopes.join(scopeDelimiter || ' ') : requestedScopes;
+  const redirectUri = makeRedirectUrl(port);
+  const callback = async (code: string): Promise<TokenCallbackResponse> => {
+    const params = {
+      grant_type: 'authorization_code',
+      code,
+      client_id: clientId,
+      client_secret: clientSecret,
```

Comment:
It looks like this implementation always passes the `client_secret` as a POST form parameter, which doesn't match the production environment. There, it first attempts to pass it as a form parameter, and if that fails tries to pass it in an `Authorization: Basic` header. While long term we may want to make this an explicit option, for now we should likely replicate the logic in prod.

Reply:
Yeah let's make sure we handle the fallback here.

Reply:
sure we can do that.